pkg/instancetype/instancetype_test.go

```diff
@@ -1093,23 +1094,318 @@ var _ = Describe("Instancetype and Preferences", func() {
+					Expect(vmi.Spec.Domain.CPU.Sockets).To(Equal(expectedCPU.Sockets))
+					Expect(vmi.Spec.Domain.CPU.Cores).To(Equal(expectedCPU.Cores))
+					Expect(vmi.Spec.Domain.CPU.Threads).To(Equal(expectedCPU.Threads))
+				},
+					Entry("by default to SocketsCores with a default topology for 1 vCPU",
+						uint32(1),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{},
+						v1.CPU{Sockets: 1, Cores: 1, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 2 vCPUs and a default ratio of 1:2:1",
+						uint32(2),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{},
+						v1.CPU{Sockets: 1, Cores: 2, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 4 vCPUs and a default ratio of 1:2:1",
+						uint32(4),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{},
+						v1.CPU{Sockets: 2, Cores: 2, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 6 vCPUs and a default ratio of 1:2:1",
+						uint32(6),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{},
+						v1.CPU{Sockets: 3, Cores: 2, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 8 vCPUs and a default ratio of 1:2:1",
+						uint32(8),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{},
+						v1.CPU{Sockets: 4, Cores: 2, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 3 vCPUs and a ratio of 1:3:1",
+						uint32(3),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1beta1.CPUPreferences{
+								SpreadOptions: &instancetypev1beta1.SpreadOptions{
+									Ratio: pointer.P(uint32(3)),
+								},
+							},
+						},
+						v1.CPU{Sockets: 1, Cores: 3, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 6 vCPUs and a ratio of 1:3:1",
+						uint32(6),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1beta1.CPUPreferences{
+								SpreadOptions: &instancetypev1beta1.SpreadOptions{
+									Ratio: pointer.P(uint32(3)),
+								},
+							},
+						},
+						v1.CPU{Sockets: 2, Cores: 3, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 9 vCPUs and a ratio of 1:3:1",
+						uint32(9),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1beta1.CPUPreferences{
+								SpreadOptions: &instancetypev1beta1.SpreadOptions{
+									Ratio: pointer.P(uint32(3)),
+								},
+							},
+						},
+						v1.CPU{Sockets: 3, Cores: 3, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 12 vCPUs and a ratio of 1:3:1",
+						uint32(12),
+						instancetypev1beta1.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1beta1.CPUPreferences{
+								SpreadOptions: &instancetypev1beta1.SpreadOptions{
+									Ratio: pointer.P(uint32(3)),
+								},
+							},
+						},
+						v1.CPU{Sockets: 4, Cores: 3, Threads: 1},
+					),
+					Entry("by default to SocketsCores with 4 vCPUs and a ratio of 1:4:1",
```

Comment:
Isn't this the same as 3 vCPUs and a ratio of `1:3:1`? (essentially vCPUs == ratio)

Same for the tests below.

Reply:
Yeah this is more to prove nothing weird is happening with an increased number of vCPUs and ratio but you're right it's basically the same. Happy to drop if you think it's just duplication.

Reply:
Let's keep it!